cmd/repo-updater/repos/syncer.go

```diff
@@ -89,69 +138,94 @@ func sleep(ctx context.Context, d time.Duration) {
 	}
 }
 
-// TriggerSync will run Sync now. If a sync is currently running it is
-// cancelled.
-func (s *Syncer) TriggerSync() {
-	s.syncSignal.Trigger()
+// TriggerEnqueueSyncJobs will enqueue any pending sync jobs now.
+func (s *Syncer) TriggerEnqueueSyncJobs() {
+	s.enqueueSignal.Trigger()
 }
 
-// Sync synchronizes the repositories.
-func (s *Syncer) Sync(ctx context.Context) (err error) {
+// SyncExternalService syncs repos using all of the supplied external services
+func (s *Syncer) SyncExternalService(ctx context.Context, store Store, externalServiceID int64, minSyncInterval time.Duration) (err error) {
 	var diff Diff
 
-	ctx, save := s.observe(ctx, "Syncer.Sync", "")
+	log15.Info("Syncing external service", "serviceID", externalServiceID)
+	ctx, save := s.observe(ctx, "Syncer.SyncExternalService", "")
 	defer save(&diff, &err)
-	defer s.setOrResetLastSyncErr(&err)
-
-	if s.FailFullSync {
-		return errors.New("Syncer is not enabled")
-	}
+	defer s.setOrResetLastSyncErr(externalServiceID, &err)
 
 	var streamingInserter func(*Repo)
 	if s.SubsetSynced == nil {
 		streamingInserter = func(*Repo) {} //noop
 	} else {
-		streamingInserter, err = s.makeNewRepoInserter(ctx)
+		streamingInserter, err = s.makeNewRepoInserter(ctx, store)
 		if err != nil {
 			return errors.Wrap(err, "syncer.sync.streaming")
 		}
 	}
 
-	var sourced Repos
-	if sourced, err = s.sourced(ctx, streamingInserter); err != nil {
-		return errors.Wrap(err, "syncer.sync.sourced")
+	ids := []int64{externalServiceID}
+	svcs, err := store.ListExternalServices(ctx, StoreListExternalServicesArgs{IDs: ids})
+	if err != nil {
+		return errors.Wrap(err, "fetching external services")
 	}
 
-	store := s.Store
-	if tr, ok := s.Store.(Transactor); ok {
-		var txs TxStore
-		if txs, err = tr.Transact(ctx); err != nil {
-			return errors.Wrap(err, "syncer.sync.transact")
-		}
-		defer txs.Done(&err)
-		store = txs
+	if len(svcs) != 1 {
+		return errors.Wrap(err, "external service not found")
```

Comment:
nit: Just to be precise... it is misleading if for whatever reason we got more than one external services here.

```suggestion
	if len(svcs) != 1 {
		return errors.Wrapf(err, "want 1 external service but got %d", len(svcs))
```